daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/SpecifiedLengthParsers.scala

```diff
@@ -90,7 +90,9 @@ sealed abstract class SpecifiedLengthParserBase(
     Assert.invariant(dis eq pState.dataInputStream)
     val bitsToSkip = finalEndPos0b - dis.bitPos0b
     Assert.invariant(bitsToSkip >= 0) // if this is < 0, then the parsing of children went past the limit, which it isn't supposed to.
-    if (bitsToSkip > 0) {
+    if (bitsToSkip > Int.MaxValue) {
+      PE(pState, "Data too large, consider using blobs instead: https://cwiki.apache.org/confluence/display/DAFFODIL/Proposal%3A+Binary+Large+Objects")
+    } else if (bitsToSkip > 0) {
```

Comment:
I'm concerned there are other places we might need to deal with these kinds of overflow issues?

For example, all the binary number parsers have a getBitLength funciton that returns an int. I would never expect a binary number to have such a large length that it overflows, but if there were bad data or a bad dfdl:length expression, it could definiately happen. If we're lucky it overflows to a negative number and hopefully we get an error message, but if it overflows to some positive number we're just going to get unexpected behavior.

I'm not sure of a good way to handle this cleanly, since we deal with lengths all over the place. Just grepping for getBitLength shows something lke 50 different implementations across different parsers and unparsers. I suspect we're not handling overflow properly in many of them. And getLengthInBits shows up as well. There are probably other length calcualtions that don't use either of these.

Reply:
Hm, would a better fix be to modify the various get functions to use Longs, but then throw a ParseError inside the get functions if the value ends up being larger than Int.MaxValue?

Reply:
I pushed up changes that I think cover all the getByteArray issues. As for the binaryNumberParsers, I think it might be worth making its own ticket that can be dealt with later as it is a much less likely situation to occur.

Reply:
Yeah, agreed this is too much to fix for now. I think we really do need to come up with a comprehensive way to ensure we dont' have any overflow issues, but it's fine to create a new bug and handle that more completely later.